src/orion/client/experiment.py

```diff
@@ -786,9 +776,6 @@ def workon(
         if not pool_size:
             pool_size = n_workers
 
-        if not reservation_timeout:
```

Comment:
It was in the function signature but it was never used, nor any deprecation warnings? :scream: I'm glad you worked on this! :smile: